chainstate/tx-verifier/src/transaction_verifier/signature_destination_getter.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::chain::{Destination, PoolId, TxOutput};
+use pos_accounting::PoSAccountingView;
+
+use super::accounting_delta_adapter::PoSAccountingDeltaAdapter;
+
+pub type SignatureDestinationGetterFn<'a> =
+    dyn Fn(&TxOutput) -> Result<Destination, SignatureDestinationGetterError> + 'a;
```

Comment:
Seems to me it should be possible to return a reference to destination here, it is always cloned in the implementation below as far as I can see. The lifetimes may be non-trivial though so likely not worth it.

Reply:
Nope. It's not possible. I already tried that yesterday. There's a destination that's pulled from the database through accounting.

Reply:
Hmm I see so `Cow` at best, not worth it.